desc/equilibrium/coords.py

```diff
@@ -376,131 +382,147 @@ def fixup(x, *args):
         )
     )
     rho, theta_PEST, zeta = coords.T
+    theta = vecroot(
+        # Assume λ=0 for default initial guess.
+        setdefault(guess, theta_PEST),
+        theta_PEST,
+        rho,
+        zeta,
+    )
     if full_output:
-        theta, (res, niter) = vecroot(
-            # Assume λ=0 for default initial guess.
-            setdefault(guess, theta_PEST),
-            theta_PEST,
-            rho,
-            zeta,
-        )
-    else:
-        theta = vecroot(
-            # Assume λ=0 for default initial guess.
-            setdefault(guess, theta_PEST),
-            theta_PEST,
-            rho,
-            zeta,
-        )
+        theta, (res, niter) = theta
     out = jnp.column_stack([rho, jnp.atleast_1d(theta.squeeze()), zeta])
     if full_output:
         return out, (res, niter)
     return out
 
 
-# TODO(#568): decide later whether to assume given phi instead of zeta.
+def _partial_sum(lmbda, L_lmn, omega, W_lmn, iota):
+    """Convert FourierZernikeBasis to set of Fourier series.
+
+    TODO(#1243) Do proper partial summation once the DESC
+    basis are improved to store the padded tensor product modes.
+    https://github.com/PlasmaControl/DESC/issues/1243#issuecomment-3131182128.
+
+    Parameters
+    ----------
+    lmbda : Transform
+        FourierZernikeBasis
+    L_lmn : jnp.ndarray
+        FourierZernikeBasis basis coefficients for λ.
+    omega : Transform
+        FourierZernikeBasis
+    W_lmn : jnp.ndarray
+        FourierZernikeBasis basis coefficients for ω.
+    iota : jnp.ndarray
+        Shape (lmbda.grid.num_rho, )
+
+    Returns
+    -------
+    lmbda_minus_iota_omega, modes
+        Spectral coefficients and modes.
+        Shape (num rho, num zeta, num modes).
+
+    """
+    grid = lmbda.grid
+    errorif(not grid.fft_poloidal, NotImplementedError, msg="See note in docstring.")
+    # TODO(#1243): assert grid.sym==eq.sym once basis is padded for partial sum
+    # TODO: (#568)
+    warnif(
+        grid.M > lmbda.basis.M,
+        ResolutionWarning,
+        msg="Poloidal grid resolution is higher than necessary for coordinate mapping.",
+    )
+    warnif(
+        grid.M < lmbda.basis.M,
+        ResolutionWarning,
+        msg="High frequency lambda modes will be truncated in coordinate mapping.",
+    )
+    lmbda_minus_iota_omega = lmbda.transform(L_lmn)
+    lmbda_minus_iota_omega = (
+        rfft(grid.meshgrid_reshape(lmbda_minus_iota_omega, "rzt"), norm="forward")
+        .at[..., (0, -1) if ((grid.num_theta % 2) == 0) else 0]
+        .divide(2)
+        * 2
+    )
+    return lmbda_minus_iota_omega, jnp.fft.rfftfreq(grid.num_theta, 1 / grid.num_theta)
+
+
+@partial(jit, static_argnames=["tol", "maxiter"])
 def _map_clebsch_coordinates(
-    coords,
     iota,
+    alpha,
+    zeta,
     L_lmn,
-    L_basis,
-    guess=None,
-    period=np.inf,
+    lmbda,
+    theta0=None,
```

Comment:
theta0 is a very bad variable name. theta0 is dangerously close to theta_0 that people in the gyrokinetics communit use in flux tube codes to define the location of vanishing integrated magnetic shear. 
Why did you change it from guess? I strongly recomment you change it back. guess is great because it immediately tells you what it is. theta0 requires user to figure out what is happening in the rest of the code.

Reply:
As a rule of thumb, please don't make unnecessary changes to variable names.

Reply:
1. This is a new function, so nothing is an unnecessary change because nothing is a change; it is new.[^1]
2. `theta0` is the initial guess in the root finding for `theta`. I will change it to `guess` for you. Whatever the name is, the public documentation next to the parameter `theta0: jnp.ndarray : Optional initial guess for the computational coordinates.` is there. So no one has to read code to figure out what it should be.

[^1]: The function `_map_clebsch_coordinates` is private and not used anywhere in DESC. If there is external code that was using this function, we are not responsible for breaking API convention.